go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -748,7 +749,11 @@ type poolConn interface {
 
 func (qre *QueryExecutor) execSQL(conn poolConn, sql string, wantfields bool) (*sqltypes.Result, error) {
 	defer qre.logStats.AddRewrittenSQL(sql, time.Now())
-	return conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	res, err := conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	if res != nil && int64(len(res.Rows)) > qre.tsv.qe.warnResultSize.Get() {
+		log.Warningf("Results returned exceeds warning threshold: %q", sql)
```

Comment:
Can you add the returned result size and warning result size to this warning? Also kind of sucks we'll lose callerid context you added to vterrors. Not sure if we can grab it from the `qre.ctx`, afaik that should be passed all the way down so might be the same context.

In terms of testing -- rather than test for a log, you could test for a counter. I had planned to use the existing Results histogram in the /debug/vars but it can't hurt to have a new counter for requests that are over the warning size. That would make some things easier, including testing. At that point you can test for the counter incrementing.